internal/upgrader/upgrader_test.go

```diff
@@ -181,55 +181,84 @@ var _ = Describe("Upgrade", func() {
 	})
 
 	When("running with --check-up-to-date", func() {
-		It("should print out service GUIDs and not attempt to upgrade", func() {
-			result := captureStdout(func() {
-				l := logger.New(100 * time.Millisecond)
-				defer l.Cleanup()
-				err := upgrader.Upgrade(fakeCFClient, l, upgrader.UpgradeConfig{
-					BrokerName:       fakeBrokerName,
-					ParallelUpgrades: 5,
-					CheckUpToDate:    true,
-				})
-				Expect(err).To(MatchError("check up-to-date failed: found 3 instances which are not up-to-date"))
-			})
+		Context("there are instances not up to date", func() {
+			var notUpToDateInstance, withPlanDeactivated ccapi.ServiceInstance
+			BeforeEach(func() {
+				fakeCFClient.GetServicePlansReturns([]ccapi.ServicePlan{
+					{
+						GUID:                   "fake-plan-1-guid",
+						Available:              true,
+						Name:                   "fake-plan-1-name",
+						MaintenanceInfoVersion: "1.5.7",
+					},
+					{
+						GUID:                   "fake-plan-2-guid",
+						Available:              true,
+						Name:                   "fake-plan-2-name",
+						MaintenanceInfoVersion: "1.5.7",
+					},
+				}, nil)
 
-			By("getting the service plans")
-			Expect(fakeCFClient.GetServicePlansCallCount()).To(Equal(1))
-			Expect(fakeCFClient.GetServicePlansArgsForCall(0)).To(Equal(fakeBrokerName))
+				fakeInstance1 = ccapi.ServiceInstance{
+					GUID:                              "fake-instance-guid-1",
+					Name:                              "fake-instance-name-1",
+					UpgradeAvailable:                  false, // no upgradable, skipped instance
+					MaintenanceInfoVersion:            "1.3.0",
+					ServicePlanGUID:                   "fake-plan-1-guid",
+					ServicePlanMaintenanceInfoVersion: "1.5.7",
+					ServicePlanDeactivated:            false,
+				}
 
-			By("getting the service instances")
-			Expect(fakeCFClient.GetServiceInstancesForServicePlansCallCount()).To(Equal(1))
-			Expect(fakeCFClient.GetServiceInstancesForServicePlansArgsForCall(0)).To(Equal([]ccapi.ServicePlan{fakePlan}))
+				fakeInstance2 = ccapi.ServiceInstance{
+					GUID:                              "fake-instance-guid-2",
+					Name:                              "fake-instance-name-2",
+					UpgradeAvailable:                  true,
+					MaintenanceInfoVersion:            "1.5.7",
+					ServicePlanGUID:                   "fake-plan-1-guid",
+					ServicePlanMaintenanceInfoVersion: "1.5.7",
+					ServicePlanDeactivated:            false,
+				}
 
-			By("not calling upgrade")
-			Expect(fakeCFClient.UpgradeServiceInstanceCallCount()).Should(Equal(0))
+				// Plan and instance with different version
+				notUpToDateInstance = ccapi.ServiceInstance{
```

Comment:
suggestion: this is an example hard to reason about. Whilst it can happen if the broker's catalog is corrupted, it is unclear to me what would the cloud controller do in this case.  We should probably look into the cloud controller logic and see how it would treat this scenario.